core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -690,7 +690,7 @@ public List<ManifestFile> apply(TableMetadata base) {
         base.schema(), current != null ? current.dataManifests() : null);
     long minDataSequenceNumber = filtered.stream()
         .map(ManifestFile::minSequenceNumber)
-        .filter(seq -> seq > 0) // filter out unassigned sequence numbers in rewritten manifests
+        .filter(seq -> seq >= 0) // filter out unassigned sequence numbers in rewritten manifests
```

Comment:
@vanliu-tx, I don't see how this would matter. Sequence number 0 is reserved for v1 files, so when you convert to v2 all of the v1 files have sequence number 0. And in v1, there are no delete files. So reporting a min sequence number of 1 and removing all delete files older than 1 should have no change.

Reply:
The check in `ManifestFilterManager` that uses this value is: `(isDelete && entry.sequenceNumber() > 0 && entry.sequenceNumber() < minSequenceNumber)`

We know that `entry.sequenceNumber()` is non-zero. It shouldn't matter whether `minDataSequenceNumber` is 0 or 1.

Reply:
Assuming the following steps
1. create v1 table and append some data files.
2. upgrade to v2 and commit.
3. append some row-delta delete and commit.
4. remove one data file and commit.
5. remove another data file and commit.

```
    long minDataSequenceNumber = filtered.stream()
        .map(ManifestFile::minSequenceNumber)
        .filter(seq -> seq > 0) // filter out unassigned sequence numbers in rewritten manifests
        .reduce(base.lastSequenceNumber(), Math::min);
```
Because all v1 data files has sequence number 0, the minDataSequenceNumber will use base.lastSequenceNumber() as it's value. In step 4 this value is 1, deleteFilterManager.dropDeleteFilesOlderThan(minDataSequenceNumber) will not drop delete-files in step 3. But in step 5, this value turns to 2, which will drop delete-files in step 3.

Reply:
Okay, I understand. The `base.lastSequenceNumber()` is what causes this. Thanks!